### ISSUE TITLE
Duplicate orders getting created for PayPal Express Checkout payment

### ISSUE THREAD

**Issue by salmanazeez786** [magento/magento2#14592]
### Preconditions

1. Magento 2.2.2
2. PayPal Express payment method is set up.
3. PayPal account having at least 2 credit cards in which 1 card is expired or blocked and the other card should be working fine.

### Steps to reproduce

1. Select PayPal Express payment method in checkout page and place order (continue to PayPal).

2. In PayPal Payment page select the expired card for payment and click on 'Pay Now' button.

3. The customer is redirected to this URL : https://www.paypal.com/checkoutnow?token=EC-9U42XXXXXXXXXXXXX (no matter if you are using In-context or not) which shows PayPal payment page again but without the expired card.( i.e it will show the other credit cards of the customer)

4. At this point an order is created in Magento  with empty state and status. (ex : 1000000001)

5. Now when customer selects the valid card to make payment in the new PayPal payment page mentioned in step 3, the payment will be successful but a new order is created  (ex : 1000000002).

6. After the IPN updates, both orders 1000000001 and 1000000002 are updated to 'Processing' state and the Customer  can see both the orders in the order history. But in Admin Orders we can see same transaction id for both orders.

### Expected result

1. Order should get created only if the PayPal payment is successful.

### Actual result

1. Orders are getting created for failed Payments  done at PayPal using declined cards
2. And when the customer tries to make payments  for same cart using all his expired cards, those number of orders are created in magento and at last when a valid payment is made then the transaction Id is assigned to all these orders and customer can see all these orders as 'Processing' state.

In below screenshots you can  2 orders 10000000501 , 10000000502 and their transaction Id is same for both orders (  0X6XXXXXXXXXXX62L ).Also the status is also same (Processing)
Order # 10000000501
![paypal-1](https://user-images.githubusercontent.com/8183885/38485296-d9798e16-3bf6-11e8-9b67-cf5c845b4d37.png)
Order # 10000000502
![paypal-2](https://user-images.githubusercontent.com/8183885/38485316-eb5bab64-3bf6-11e8-9be1-e5194acc419f.png)

**Comment by supplyreb** [comment]
We are on M 2.2.3 and still have this issue.

**Comment by ghost** [comment]
Hi @salmanazeez786 i'm not able to reproduce by following steps your described, i'm not able to add expired credit card it's impossible to testing.

**Comment by ghost** [comment]
@salmanazeez786 , we are closing this issue due to inactivity. If you'd like to update it, please reopen the issue.

**Comment by salmanazeez786** [comment]
> Hi @salmanazeez786 i'm not able to reproduce by following steps your described, i'm not able to add expired credit card it's impossible to testing.

You cannot add expired credit cards,but if you had previously added a card and that got expired recently,then this will occur.

**Comment by ghost** [comment]
@salmanazeez786 I'm not  able to reproduce, because i'm not able to have previously added a card and that got expired.

**Comment by sezio** [comment]
I think it is still occuring in 2.2.5.
Can you maybe ask paypal to add an expired credit card in a sandbox account ? 

I don't get why it's closed as it is a bug, no ? not being able to add an expired card does not mean the problem does not exists ?

**Comment by charleskj** [comment]
We had experience the same issue, this is still existing

**Comment by skynixukraine** [comment]
Hello,
We discovered this issue a long time ago when Magento 2 was born.
As Magento team ignores this issue, we have coded own fix for this.
Everyone welcome to get it for free https://skynix.company/blog/solutions/duplicated-paypal-orders-in-magento-2-free-fix/

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
Hi @salmanazeez786 i'm trying to reproduce this, i have expired credit cart, but this cart not appear in list when i whant to pay. And i try to do another way. -> https://developer.paypal.com/docs/classic/api/errors/?mark=api%20error%20codes#10500-to-10599 
so i also have no duplicated orders. How else i can reproduce this ?

**Comment by ghost** [comment]
HI @salmanazeez786  we are closing this issue due to inactivity. If you'd like to update it, please reopen the issue.

**Comment by devendra-nationalwala** [comment]
We have encountered this issue in Magento ver. 2.2.5 Enterprise edition.

**Comment by adam-inco** [comment]
Hi, This issue is also in Magento 2.3 hoping skynixukraine fix will solve this as it seems it cannot be tested easily without an expired card :S

**Comment by fmjrey** [comment]
This is a duplicate of #13952
I can confirm this also happens with successful payments.

**Comment by Deepeey** [comment]
Hello. Is there any updates on this? I'm experiencing duplicated orders using paypal on my magento 2.2.8. Can you help? 
Already done with the skynix but still no luck. Thanks.
